src/lucky/events/pipe_event.cr

```diff
@@ -0,0 +1,15 @@
+class Lucky::Events::PipeEvent < Pulsar::Event
```

Comment:
I originally debated on if this should be `BeforePipeEvent` and `AfterPipeEvent`, or just make them the same and pass through the position of the pipe... I'm open to other naming options if this sounds confusing.